flw/src/main/java/org/motechproject/nms/flw/service/impl/FrontLineWorkerServiceImpl.java

```diff
@@ -150,6 +163,69 @@ public State getState(FrontLineWorker frontLineWorker) {
         return state;
     }
 
+    @Override // NO CHECKSTYLE Cyclomatic Complexity
+    public boolean createUpdate(Map<String, Object> flw) {
+
+        long stateId = (long) flw.get(FlwConstants.STATE_ID);
+        long districtId = (long) flw.get(FlwConstants.DISTRICT_ID);
+        String mctsFlwId = flw.get(FlwConstants.ID).toString();
+        long contactNumber = (long) flw.get(FlwConstants.CONTACT_NO);
+
+        State state = locationService.getState(stateId);
+        if (state == null) {
+            flwErrorDataService.create(new FlwError(mctsFlwId, stateId, districtId, FlwErrorReason.INVALID_LOCATION_STATE));
+            return false;
+        }
+        District district = locationService.getDistrict(stateId, districtId);
+        if (district == null) {
+            flwErrorDataService.create(new FlwError(mctsFlwId, stateId, districtId, FlwErrorReason.INVALID_LOCATION_DISTRICT));
+            return false;
+        }
+
+        FrontLineWorker existingFlwByNumber = getByContactNumber(contactNumber);
+        FrontLineWorker existingFlwByMctsFlwId = getByMctsFlwIdAndState(mctsFlwId, state);
+        Map<String, Object> location = new HashMap<>();
+        try {
+            location = locationService.getLocations(flw);
+
+            if (existingFlwByMctsFlwId != null && existingFlwByNumber != null) {
+
+                if (existingFlwByMctsFlwId.getMctsFlwId().equalsIgnoreCase(existingFlwByNumber.getMctsFlwId()) &&
+                        existingFlwByMctsFlwId.getState().equals(existingFlwByNumber.getState())) {
+                    // we are trying to update the same existing flw. set fields and update
+                    LOGGER.debug("Updating existing user with same phone number");
+                    update(FlwMapper.updateFlw(existingFlwByMctsFlwId, flw, location));
+                    return true;
+                } else {
```

Comment:
That's good catch. Technically, I do need to catch all anonymous(existing) users without an mcts id and assign them the mcts update with the same phone number. **Need to fix**